### PR TITLE
recommendation function extended for future flag

### PR DESCRIPTION
**Pull request by angsherpa232**

### CODE REVIEW

File: src/recommend.js [hunk 1]
```diff
@@ -1,34 +1,55 @@
 import recommendations from '../data/recommendations.json';
+import types from '../data/types.json';
 
 function recommendTreeSpecies(forestType) {
+  if (types.forestType.find(v => v.code === forestType) === undefined) {
+    throw new Error(`${forestType} is not a valid forest Type`);
```
Comment (friedjoff) [inline src/recommend.js]:
```suggestion
    throw new Error(`${forestType} is not a valid forest type`);
```

File: src/recommend.js [hunk 2]
```diff
@@ -1,34 +1,55 @@
...
     throw new Error(
       `at least one forest type is required to get the recommendation of tree species`,
     );
-  } else if (forestType.length === 1) {
-    result = recommendTreeSpecies(forestType[0]);
-  } else if (forestType.length === 2) {
-    const { one, two, three } = recommendTreeSpecies(forestType[0]);
+  }
+
+  if (future && typeof future !== 'boolean') {
```
Comment (friedjoff) [inline src/recommend.js]:
```suggestion
  if (typeof future !== 'boolean') {
```

File: src/recommend.js [hunk 3]
```diff
@@ -1,34 +1,55 @@
...
+    throw new Error(`expected boolean type for future flag`);
+  }
+
+  let result;
+
+  result = recommendTreeSpecies(forestType1);
```
Comment (friedjoff) [inline src/recommend.js]:
```suggestion
  let result = recommendTreeSpecies(forestType1);
```
And remove line 31

File: src/recommend.test.js [hunk 4]
```diff
@@ -1,34 +1,62 @@
...
-      227200,
-      363700,
-      25300,
-      174200,
-      317500,
-      402200,
-    ],
+  test('invalid argument type for future flag', () => {
+    expect(() => recommend('60', '963D')).toThrowError(
+      `963D is not a valid forest Type`,
```
Comment (friedjoff) [inline src/recommend.test.js]: This seems wrong to me.

File: src/recommend.test.js [hunk 5]
```diff
@@ -1,34 +1,62 @@
...
+        363700,
+        25300,
+        174200,
+        317500,
+        402200,
+      ],
+    });
+  });
+
+  test('invalid input type of future flag', () => {
```
Comment (friedjoff) [inline src/recommend.test.js]: duplicated

File: src/recommend.js (hunk 2 above)
Comment (angsherpa232) [reply src/recommend.js]: if there is no check for the "future" then the execution stops at line 27 and throws the error while running other tests.

Comment (friedjoff) [reply src/recommend.js]: Ah, ok, makes sense.

File: src/recommend.test.js (hunk 4 above)
Comment (angsherpa232) [reply src/recommend.test.js]: yeah, thanks for catching it. Now fixed.

**Pull request merged**